### DISCUSSION TITLE
Is there way to integrate with my existing glutin-based app?

### DISCUSSION THREAD

**Discussion by kuy** [emilk/egui#730] [Q&A] [answered]
Hi egui community, I'm working on an app using glutin crate and want to use egui for UIs of it. But I noticed that egui (egui_glium) launches dedicated winit's EventLoop in main thread and doesn't expose it. Because my app already launches winit's EventLoop in  main thread, the two conflicts and couldn't run.

So I'm finding a way to use egui in EventLoop of my app. Do you know anyone?

P.S. I found ["Integrating with egui"](https://docs.rs/egui/0.14.2/egui/#integrating-with-egui) on docs but it is slightly abstract for me.

**Comment by emilk** [comment, marked as answer]
You can use `egui_glium` with your own even loop: https://github.com/emilk/egui/blob/master/egui_glium/examples/pure.rs

**Reply by kuy** [reply]
Thanks! I didn't see an example of `egui_glium`.
